### PR TITLE
indexer, graphql: remove pg_integration feature flag usage

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-graphql-rpc/src/test_infra/cluster.rs [hunk 1]
```diff
@@ -33,8 +33,8 @@ use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 use tracing::info;
 
-const VALIDATOR_COUNT: usize = 7;
-const EPOCH_DURATION_MS: u64 = 15000;
+const VALIDATOR_COUNT: usize = 4;
```
Comment (lxfind) [inline crates/sui-graphql-rpc/src/test_infra/cluster.rs:36]: Why this change?

File: crates/sui-graphql-e2e-tests/tests/tests.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@ datatest_stable::harness!(run_test, TEST_DIR, r".*\.(mvir|move)$");
 #[cfg_attr(msim, msim::main)]
 async fn run_test(path: &Path) -> Result<(), Box<dyn std::error::Error>> {
     telemetry_subscribers::init_for_testing();
-    if cfg!(feature = "pg_integration") {
+    if !cfg!(msim) {
```
Comment (lxfind) [inline crates/sui-graphql-e2e-tests/tests/tests.rs:19]: Can you add a comment on this `if`? Don't quite understand it.

File: crates/sui-graphql-rpc/src/test_infra/cluster.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-graphql-rpc/src/test_infra/cluster.rs:36]: The default for all of our other tests is 4, so 7 just results in more load during the test for little value since we aren't even interested in testing the validator portion of the stack. I found that this helped reduced timeouts when these tests were run when the GitHub actions runner was under high load.

File: crates/sui-graphql-e2e-tests/tests/tests.rs (hunk 2 above)
Comment (bmwill) [reply crates/sui-graphql-e2e-tests/tests/tests.rs:19]: Yeah, essentially these tests don't really work under msim since they rely on external things like a database so this shuts them off during those tests.

**Review by lxfind — APPROVED** [review]
